src/main/java/pl/agasior/interviewprep/controllers/QuestionController.java

```diff
@@ -20,20 +22,25 @@
     private final QuestionCreator questionCreator;
     private final QuestionUpdater questionUpdater;
     private final QuestionReader questionReader;
+    private final TagCreator tagCreator;
 
-    QuestionController(final QuestionCreator questionCreator, final QuestionUpdater questionUpdater, final QuestionReader questionReader) {
+    QuestionController(final QuestionCreator questionCreator, final QuestionUpdater questionUpdater,
+                       final QuestionReader questionReader, final TagCreator tagCreator) {
         this.questionCreator = questionCreator;
         this.questionUpdater = questionUpdater;
         this.questionReader = questionReader;
+        this.tagCreator = tagCreator;
     }
 
     @PostMapping
-    ResponseEntity<CreateQuestionResult> createQuestion(@RequestBody @Valid CreateQuestionCommand command) {
+    public ResponseEntity<CreateQuestionResult> createQuestion(@RequestBody @Valid CreateQuestionCommand command) {
+        //Add unique tags from command
+        command.getTags().stream().map(TagDto::new).forEach(tagDto -> tagCreator.createIfAbsent(Tag.fromDto(tagDto)));
```

Comment:
Dude, why u map all tags to dto just to map them back to entity in forEach?
Also, controllers shouldn't containt any kind of logic, I'd suggest to move that inside QuestionCreator class and place it after successful validation of Question, because now someone could send null tags set which will cause NullPointerException